src/components/accessRequests/accessRequests.tsx

```diff
@@ -76,26 +115,32 @@ const AccessRequests = () => {
           });
         }
       }
-      setRequests(requests);
+      if (accessRequests !== null) {
+        setRequests(accessRequests);
+        if (!filteredRequests)
+          setFilteredRequests(
+            accessRequests.filter((x) => x.status === "pending")
+          );
+      }
     }
-    fetchData();
-  }, [showMessage, user]);
+    if (requests === null || requests === undefined) fetchData();
```

Comment:
You can fetchData() unconditionally here. The method should only run once when the component is created, not every time it renders.